### PR TITLE
Removing special chars from str flags

### PR DESCRIPTION
**Pull request by mrpaja**
This pull request references issues #8596
For instance, instead from Hello World\n printing str.Hello_World_n it prints now 
str.Hello_World 
The code also "beautifies"  string flags (str) by removing any trailing _

### CODE REVIEW

**Comment by mrpaja** [comment]
Some tests will fail because strings are altered by removing \t, \n and so on. I dont like the leading and trailing underscores _ either. How to proceed?

**Comment by Maijin** [comment]
Fixing the test, create a Pull-Request on r2r that fix those tests accordingly to your fixes

File: libr/util/name.c [hunk 1]
```diff
@@ -40,12 +45,22 @@ R_API int r_name_filter(char *name, int maxlen) {
...
 			//		r_str_ccpy (name, name+1, 0);
 			//name--;
 		}
 	}
+	while (i > 0) {
+		if (*(name - 1) == '\\' && is_special_char (name)) {
+			*name = '_';
+			*(name - 1) = '_';
+		}
+		--name;
```
Comment (radare) [inline libr/util/name.c]: dont predecrement unless strictly necesary. just use the post-dec

File: libr/util/name.c [hunk 2]
```diff
@@ -27,6 +27,11 @@ R_API int r_name_check(const char *name) {
 	return true;
 }
 
+static inline bool is_special_char (char *name) {
+	return (*name == 'b' || *name == 'f' || *name == 'n' || \
```
Comment (radare) [inline libr/util/name.c]: const char n == *name;
return (n=='b' || n=='f' ...

**Comment by radare** [comment]
cc @kazarmy

**Comment by mrpaja** [comment]
Test were adjusted in this https://github.com/radare/radare2-regressions/pull/1090

**Comment by kazarmy** [comment]
> Test were adjusted in this radare/radare2-regressions#1090

What happened to removal of trailing and leading underscores?

**Comment by mrpaja** [comment]
Don't know, I didn't touch that. As you see, I just added code.

**Comment by kazarmy** [comment]
Have to add more code then, since saying that `str.Hello_World__` is better than `str.Hello_World_n` doesn't seem defensible. It's not that difficult.

**Comment by mrpaja** [comment]
Thanks for feedback. Gonna add that!

**Comment by kazarmy** [comment]
What I **_suggest_** (if you need any tips) is:

 1. Removing trailing underscores just involves strategically placing a `\0` somewhere in the flag. I think you did this and then backed out (?!?), as evidenced by d4bb32880d0b91426b683410fe12aa0a26fddc27.
 1. Removing leading underscores should involve a `memmove`, **not** `memcpy`. There are (a lot of) examples in r2's code regarding `memmove`. 
    1. There is no need to worry about memory management since the flag isn't growing.
    1. There shouldn't be much of a performance hit since the flag should be in a CPU cache somewhere.
 1. Remove trailing underscores first.

Note that these are just **_suggestions_** since I haven't written any code for this.

**Comment by mrpaja** [comment]
Thx for pointing out viable ways to improve the patch. I will definitely check them first. Great learning experience!

**Comment by mrpaja** [comment]
Tests will fail because of removing leading and trailing underscores. There is a pull request radare/radare2-regressions#1090 fixing the tests.

**Comment by kazarmy** [comment]
I'm currently unwell and not able to review the code. Might take a few more days for me to be well again.

**Comment by mrpaja** [comment]
Get well soon!

File: libr/util/name.c [hunk 3]
```diff
@@ -40,12 +45,35 @@ R_API int r_name_filter(char *name, int maxlen) {
...
+		}
+		name--;
+		i--;
+	}
+	if (*name == '\\') {
+		*name = '_';
+	}
+	// trimming trailing and leading underscores
+	len = strlen (name);
+	for (len; *(name + len - 1) == '_'; len--);
```
Comment (kazarmy) [inline libr/util/name.c]: Just use
```
for (;
```
That non-initialization of `len` can be misread as a check of `len` against 0.

File: libr/util/name.c [hunk 4]
```diff
@@ -40,12 +45,35 @@ R_API int r_name_filter(char *name, int maxlen) {
...
+	if (!len) { // name conists only of underscore
```
Comment (kazarmy) [inline libr/util/name.c]: conists -> consists
underscore -> underscores

**Pull request merged**